surefire-grouper/pom.xml

```diff
@@ -32,18 +32,53 @@
   <name>Maven Surefire Test-Grouping Support</name>
   <description>Maven Surefire Test-Grouping Support</description>
 
+  <properties>
+    <javacc.version>7.0.10</javacc.version>
+    <generated.sources.dir>${project.basedir}/src/generated/java</generated.sources.dir>
+  </properties>
+
   <build>
     <plugins>
       <plugin>
-        <groupId>org.codehaus.mojo</groupId>
+        <groupId>org.javacc.plugin</groupId>
         <artifactId>javacc-maven-plugin</artifactId>
-        <version>2.6</version>
+        <version>3.0.3</version>
         <executions>
           <execution>
             <id>javacc</id>
             <goals>
               <goal>javacc</goal>
             </goals>
+            <configuration>
+              <sourceDirectory>${project.basedir}/src/main/javacc</sourceDirectory>
+              <outputDirectory>${generated.sources.dir}</outputDirectory>
```

Comment:
why change default value .... `${project.build.directory}/generated-sources/javacc`

Reply:
It's a bad default imho. People don't expect source folders to exist in the build directory.

Reply:
@sman-81 
Don;t modify default settings unless it is absolutely necessary.
The old plugin is compliant with IDEs, the default layout, where generates sources appear in `target/generated-sources/javacc`.